torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,454 @@
+                {'id': df['id'].iloc[0], 'seq_id': list(range(i, i + k))}
+                for i in range(min_seq_id, max_seq_id - k + 2)
+            ]
+            return [
+                subseq
+                for subseq in subsequences
+                if set(subseq['seq_id']).issubset(df['seq_id'])
+            ]
+
+        self.sample_sequences: list[_SampleSequenceDict] = [
+            item
+            for sublist in self.aux_df.groupby('id')
+            .apply(_get_subsequences, k=self.sequence_length)
+            .tolist()
+            for item in sublist
+        ]
+
+    def __getitem__(self, index: int) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data, labels, and metadata at that index
+        """
+        sample_entry = self.sample_sequences[index]
+        sample_df = self.aux_df[
+            (self.aux_df['id'] == sample_entry['id'])
+            & (self.aux_df['seq_id'].isin(sample_entry['seq_id']))
+        ]
+
+        sample = {'image': self._load_image(sample_df)}
+        # load features of the last image in the sequence
+        sample.update(
+            self._load_features(
+                os.path.join(
+                    self.root,
+                    self.data_root,
+                    'metadata',
+                    str(sample_df.iloc[-1]['id']) + '.csv',
+                ),
+                sample_df.iloc[-1]['image_path'],
+            )
+        )
+
+        # torchgeo expects a single label
+        sample['label'] = torch.Tensor([sample[target] for target in self.targets])
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.sample_sequences)
+
+    def _load_image(self, sample_df: pd.DataFrame) -> Tensor:
+        """Load a single image.
+
+        Args:
+            sample_df: df holding all information necessary to load the
+                consecutive images in the sequence
+
+        Returns:
+            concatenation of all images in the sequence over channel dimension
+        """
+
+        def load_image_tensor(id: str, filepath: str) -> Tensor:
+            """Load a single image tensor from a h5 file.
+
+            Args:
+                id: typhoon id
+                filepath: path to the h5 file
+
+            Returns:
+                image tensor
+            """
+            h5py = lazy_import('h5py')
+
+            full_path = os.path.join(self.root, self.data_root, 'image', id, filepath)
+            with h5py.File(full_path, 'r') as h5f:
+                # tensor with added channel dimension
+                tensor = torch.from_numpy(h5f['Infrared'][:]).unsqueeze(0)
+
+                # follow normalization procedure
```

Comment:
Wondering if this should be in the datamodule instead. Possibly also true for other stuff.

Reply:
I think it has to be part of the dataset because the normalization values can change. In the datamodule it will become quiet messy, and overall I think it's much nicer if normalization happens in the dataset and any augmentations in the datamodule.

Reply:
Only complaint is that the user can't perform their own normalization, or if they do it is now duplicated. Definitely something worth discussing more broadly though (not just in your PR, for all datasets).